hldb/Lib/DirectConnector.swift

```diff
@@ -9,24 +9,460 @@
+    
+    
+  }
+  
+  func bind(obj: AnyObject?, atIndex index:Int, inStatement pStmt:COpaquePointer) -> Int32 {
+    let idx = Int32(index)
+    
+    if obj == nil {
+      return sqlite3_bind_null(pStmt, Int32(idx))
+    } else if let blobObj = obj as? NSData {
+      return sqlite3_bind_blob(pStmt, idx, blobObj.bytes, Int32(blobObj.length), SQLITE_TRANSIENT)
+    } else if let stringObj = obj as? String {
+      return sqlite3_bind_text(pStmt, idx, stringObj, Int32(stringObj.characters.count), SQLITE_TRANSIENT)
+    } else if let int8Obj = obj as? Int8 {
+      return sqlite3_bind_int(pStmt, idx, Int32(int8Obj))
+    } else if let uint8Obj = obj as? UInt8 {
+      return sqlite3_bind_int(pStmt, idx, Int32(uint8Obj))
+    } else if let int16Obj = obj as? Int16 {
+      return sqlite3_bind_int(pStmt, idx, Int32(int16Obj))
+    } else if let uint16Obj = obj as? UInt16 {
+      return sqlite3_bind_int(pStmt, idx, Int32(uint16Obj))
+    } else if let int32Obj = obj as? Int32 {
+      return sqlite3_bind_int(pStmt, idx, Int32(int32Obj))
+    } else if let uint32Obj = obj as? UInt32 {
+      return sqlite3_bind_int(pStmt, idx, Int32(uint32Obj))
+    } else if let intObj = obj as? Int {
+      return sqlite3_bind_int(pStmt, idx, Int32(intObj))
+    } else if let uintObj = obj as? UInt {
+      return sqlite3_bind_int64(pStmt, idx, Int64(uintObj))
+    } else if let int64Obj = obj as? Int64 {
+      return sqlite3_bind_int64(pStmt, idx, int64Obj)
+    } else if let uint64Obj = obj as? UInt64 {
+      return sqlite3_bind_int64(pStmt, idx, Int64(uint64Obj))
+    } else if let floatObj = obj as? Float {
+      return sqlite3_bind_double(pStmt, idx, Double(floatObj))
+    } else if let float80Obj = obj as? Float80 {
+      return sqlite3_bind_double(pStmt, idx, Double(float80Obj))
+    } else if let doubleObj = obj as? Double {
+      return sqlite3_bind_double(pStmt, idx, doubleObj)
+    } else if let obj = obj {
+      fatalError("tried to bind unexpected value \(obj)")
+    }
+    return 0
+  }
+  
+  public func lastErrorMessage() -> String {
+    return self.errorMessage
+  }
+  
+  public func lastErrorCode() -> Int {
+    return 0
+  }
+  
+  
+  func rollback() -> Bool {
+    let b = self.executeUpdate("rollback transaction")
+    
+    if (b) {
+      self.inTransaction = false
+    }
+    
+    return b
+  }
+  
+  func commit() -> Bool {
+    let b =  self.executeUpdate("commit transaction")
+    
+    if (b) {
+      self.inTransaction = false
+    }
+    
+    return b
+  }
+  
+  func beginDeferredTransaction() -> Bool {
+    
+    let b = self.executeUpdate("begin deferred transaction")
+    if (b) {
+      self.inTransaction = true
+    }
+    
+    return b
+  }
+  
+  func beginTransaction() -> Bool {
+    
+    let b = self.executeUpdate("begin exclusive transaction")
+    if (b) {
+      self.inTransaction = true
+    }
+    
+    return b;
+  }
+  
+  func isInTransaction() -> Bool {
+    return self.inTransaction;
+  }
```

Comment:
`inTransaction` should have a computed `getter` property instead of a `get` method (the former is more Swifty, the latter is more Java-y)